File: scala/space-age/src/main/scala/SpaceAge.scala
import SpaceAge._

class SpaceAge(val seconds: Seconds) {

  lazy val onEarth: Age = age(1)

  private def age(orbitalPeriod: OrbitalPeriod): Age = {
    def roundOnSecondDecimal(d: Double): Double = math.round(d * 100d) / 100d

    roundOnSecondDecimal(seconds / orbitalPeriod / EarthYear)
  }

  lazy val onMercury: Age = age(0.2408467)

  lazy val onVenus: Age = age(0.61519726)

  lazy val onMars: Age = age(1.8808158)

  lazy val onJupiter: Age = age(11.862615)

  lazy val onUranus: Age = age(84.016846)

  lazy val onSaturn: Age = age(29.447498)

  lazy val onNeptune: Age = age(164.79132)
}

object SpaceAge {
  type Seconds = Long
  type Age = Double
  type OrbitalPeriod = Double

  def apply(age: Seconds) = new SpaceAge(age)

  val EarthYear: Seconds = 31557600
}

File: java/space-age/src/main/java/SpaceAge.java
public class SpaceAge {

    private final long seconds;

    public SpaceAge(int seconds) {
        this((long)seconds);
    }

    public SpaceAge(long seconds) {
        this.seconds = seconds;
    }

    private static final long EarthYear = 31557600L;

    private double age(double orbitalPeriod) {
      return roundOnSecondDecimal(seconds / orbitalPeriod / EarthYear);
    }

    private double roundOnSecondDecimal(double d) {
        return Math.round(d * 100d) / 100d;
    }

    public long getSeconds() { return seconds; }

    public double onEarth() { return age(1); }

    public double onMercury() { return age(0.2408467); }

    public double onVenus() { return age(0.61519726); }

    public double onMars() { return age(1.8808158); }

    public double onJupiter() { return age(11.862615); }

    public double onSaturn() { return age(29.447498); }

    public double onUranus() { return age(84.016846); }

    public double onNeptune() { return age(164.79132); }
}
